neuralpredictors/layers/cores/base.py

```diff
@@ -1,24 +1,16 @@
-import warnings
-from collections import OrderedDict, Iterable
-
 from torch import nn
-import torch
-import torchvision
-
-from .. import regularizers
-from . import Bias2DLayer, Scale2DLayer
-from .activations import AdaptiveELU
-from .hermite_layers import (
-    HermiteConv2D,
-    RotationEquivariantBatchNorm2D,
-    RotationEquivariantBias2DLayer,
-    RotationEquivariantScale2DLayer,
-)
 
 
 class Core:
+    """
+    Base class for the core models where shared features (across neurons) are computed.
+    """
+
     def initialize(self):
-        raise NotImplementedError("Not initializing")
+        """
+        Initialization applied on the core.
+        """
+        raise NotImplementedError()
```

Comment:
Agreed - could you may be make it like `raise NotImplementedError('initialize method must be implemented by the inheriting class')`?